docs/rules/rules.tex

```diff
@@ -0,0 +1,1526 @@
+
+Let us fix $N∈ℕ$ and $s₀∈S$. And define $B = \min_{n∈ℕ}N < 2^{kn}$,
+and $(s_n)_{n∈ℕ}$ and $(u_n)_{n>0}$ as\improvement{I could define
+  generators once and for all and these could derive from them. This
+  seems to require a dependent product though. We can work around this
+  defining a family of generators instead.}
+\begin{itemize}
+\item $s_0$ is as given above
+\item $(u_{n+1}, s_{n+1}) = g(B)(s_n)$
+\end{itemize}
+
+Let $n₀ = \min_{n>0} \left(\rem{2^{kB}}{N} ⩽ u_n < 2^{kB}\right)$.
+
+We define
+$$
+f(N)(s₀) = (\rem{u_{n₀}}{N}, s_{n₀})
+$$
+
+Note that, by definition of the remainder,
+$\{ p∈ℕ | \rem{2^{kB}}{N} ⩽ u_n < 2^{kB} \}$ has $qN$ elements for
+some $q$. From which we can deduce that $f(N)$ is surjective on its
+first component (and actually: if the frequency of each $p$ in $u_n$
+is the same, then the same property also holds of $f$).
+
+\subsection{Chain quality}
+\label{sec:chain-quality}
+
+The chain qualitity function is defined as
+$$
+\quality{n} = 2n
+$$
+\subsection{Secret-sharing phases}
+\label{sec:vss-phases}
+
+The secret-sharing phases are defined as:
+\begin{itemize}
+\item $\commitphase = \{ n∈ℕ~|~0 ⩽ n < \quality{k} \}$
+\item $\openphase = \{ n∈ℕ~|~2×\quality{k} ⩽ n < 3×\quality{k} \}$
+\item $\recoveryphase = \{ n∈ℕ~|~4×\quality{k} ⩽ n < 5×\quality{k} \}$
+\end{itemize}
+
+\section{Issuing a block}
+\label{sec:extending-blockchain}
+
+\info{In this section we will specify the rules used by a slot-leader
+  to create a new block. The state is a single chain, plus outstanding
+  transactions.}
+
+In this section we are parametrised by the current slot, a valid chain
+$b₁,…,b_n$, as well as the agent id $a$ of the node attempting to issue
+the block (or rather \emph{an} agent id, as it is customary in Cardano
+to have many public keys for a single agent).
+
+In order to issue a new block check that:
+\begin{description}
+\item[Authorised] Let $s$ be the current slot, and $l$ be the slot
+  leader at $s$:
+  $$
+  l = \slotleader{\epochdistribution{b_1,…,b_n}}{\epochseed{b_1,…,b_n}}{\epochslot{s}}
+  $$
+
+  Then, $a$ must be transitively authorised by $l$ in that:
+  $$
+  i = \transitivefix{\delegations{b₁,…,b_n}}(l)
+  $$
+\end{description}
+
+And build the following block:
+\begin{record}
+\item[predecessor] The predecessor is the (block id of) $b_n$
+\item[slot number] The slot number is set to the current slot
+\item[issuer] The block issuer is $a$
+\item[transactions] The list of transactions is set to be the entire
+  mempool\improvement{In fact this is limited by the block size
+    somehow. See
+    \href{https://iohk.myjetbrains.com/youtrack/issue/CDEC-170}{CDEC-170}.}
+\item[delegation certificates] \improvement{Specify where the delegation
+    certificates come from.}
+\end{record}
+\improvement{TODO: updates, ssc messages, and where they come from}
+\improvement{TODO: block size limits}
+\improvement{Should we specify the state of the mempool, and the
+  various message pools as well?}
```

Comment:
I think we should go for these three `improvement`s.

Reply:
I'll be working on these, then.